chef/cookbooks/salt/recipes/role_salt_ssh.rb

```diff
@@ -0,0 +1,74 @@
+#
+# Copyright 2018, SUSE LINUX GmbH
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+if CrowbarRoleRecipe.node_state_valid_for_role?(node, "salt", "salt-ssh")
+  # requires repos
+  # http://download.nue.suse.com/ibs/SUSE/Products/SLE-Module-Adv-Systems-Management/12/x86_64/product/
+  # http://download.nue.suse.com/ibs/SUSE/Updates/SLE-Module-Adv-Systems-Management/12/x86_64/update/
+  package "salt-ssh" do
+    action :install
+  end
```

Comment:
Please don't put the real content of recipe here, and create another recipe that gets included from here.

The idea is to keep these role recipes extremely simple, so that we can add more orchestration logic easily to each of them.